spec/mysql2/result_spec.rb

```diff
@@ -410,8 +410,13 @@
     end
 
     it "should raise an error given an invalid DATETIME" do
-      expect { @client.query("SELECT CAST('1972-00-27 00:00:00' AS DATETIME) as bad_datetime").each }.to \
-        raise_error(Mysql2::Error, "Invalid date in field 'bad_datetime': 1972-00-27 00:00:00")
+      if @client.info[:version] < "8.0"
```

Comment:
I seems that this part and another part on this PR to compare the MySQL client version will not work when the MySQL 10.0 is released in the future. The following way could be the solution.

```
if (@client.info[:version].split('.').map(&:to_i) <=> [8, 0]) < 0
```

The version is about MySQL not about Gem's, but the following way could be also possible technically.

```
if Gem::Version.new(@client.info[:version]) < Gem::Version.new("8.0")-
```

I have not tested it by myself.

Reply:
To best honest it's test code, so it's not like it's gonna break apps updating or anything and we can always update it when we add MySQL 10 to the CI in 4 or 5 years.

But yeah, feel free to PR the `Gem::Version` one.